transport/src/main/java/io/netty/channel/embedded/EmbeddedEventLoop.java

```diff
@@ -65,28 +66,47 @@ public EventLoop next() {
     public void execute(Runnable command) {
         requireNonNull(command, "command");
         tasks.add(command);
+        if (!running) {
+            runTasks();
+        }
     }
 
     void runTasks() {
-        for (;;) {
-            Runnable task = tasks.poll();
-            if (task == null) {
-                break;
+        boolean wasRunning = running;
```

Comment:
To confirm, we choose to run even when already running so that existing users of EmbeddedChannel.runPendingTasks() will see all the tasks run, even if they are on the event loop.

Reply:
yes